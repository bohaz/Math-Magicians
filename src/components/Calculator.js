import { useState } from 'react';
import './styles.css';
import calculate from '../logic/calculate';

function Calculator() {
  const [currentOperand, setCurrentOperand] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleClick = (buttonName) => {
    setCurrentOperand((prevOperand) => calculate(prevOperand, buttonName));
  };

  return (
    <div className="calculator-container">
      <h2 className="clalculator-title">Let&apos;s do some math!</h2>
      <div className="calculator-grid">
        <div className="output">
          <div className="current-operand" data-testid="current-operand">{currentOperand.next || currentOperand.total || '0'}</div>
        </div>
        <button type="button" onClick={() => handleClick('AC')}>AC</button>
        <button type="button" onClick={() => handleClick('+/-')}>+/-</button>
        <button type="button" onClick={() => handleClick('%')}>%</button>
        <button type="button" className="orange" onClick={() => handleClick('รท')}>รท</button>
        <button type="button" onClick={() => handleClick('7')}>7</button>
        <button type="button" onClick={() => handleClick('8')}>8</button>
        <button type="button" onClick={() => handleClick('9')}>9</button>
        <button type="button" className="orange" onClick={() => handleClick('x')}>x</button>
        <button type="button" onClick={() => handleClick('4')}>4</button>
        <button type="button" onClick={() => handleClick('5')}>5</button>
        <button type="button" onClick={() => handleClick('6')}>6</button>
        <button type="button" className="orange" onClick={() => handleClick('-')}>-</button>
        <button type="button" onClick={() => handleClick('1')}>1</button>
        <button type="button" onClick={() => handleClick('2')}>2</button>
        <button type="button" onClick={() => handleClick('3')}>3</button>
        <button type="button" className="orange" onClick={() => handleClick('+')}>+</button>
        <button type="button" className="span-two" onClick={() => handleClick('0')}>0</button>
        <button type="button" onClick={() => handleClick('.')}>.</button>
        <button type="button" className="orange" onClick={() => handleClick('=')}>=</button>
      </div>
    </div>
  );
}

export default Calculator;
