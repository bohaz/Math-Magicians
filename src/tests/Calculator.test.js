import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Calculator from '../components/Calculator';

test('renders Calculator correctly', () => {
  const { asFragment } = render(<Calculator />);
  expect(asFragment()).toMatchInlineSnapshot(`
<DocumentFragment>
  <div
    class="calculator-container"
  >
    <h2
      class="clalculator-title"
    >
      Let's do some math!
    </h2>
    <div
      class="calculator-grid"
    >
      <div
        class="output"
      >
        <div
          class="current-operand"
          data-testid="current-operand"
        >
          0
        </div>
      </div>
      <button
        type="button"
      >
        AC
      </button>
      <button
        type="button"
      >
        +/-
      </button>
      <button
        type="button"
      >
        %
      </button>
      <button
        class="orange"
        type="button"
      >
        รท
      </button>
      <button
        type="button"
      >
        7
      </button>
      <button
        type="button"
      >
        8
      </button>
      <button
        type="button"
      >
        9
      </button>
      <button
        class="orange"
        type="button"
      >
        x
      </button>
      <button
        type="button"
      >
        4
      </button>
      <button
        type="button"
      >
        5
      </button>
      <button
        type="button"
      >
        6
      </button>
      <button
        class="orange"
        type="button"
      >
        -
      </button>
      <button
        type="button"
      >
        1
      </button>
      <button
        type="button"
      >
        2
      </button>
      <button
        type="button"
      >
        3
      </button>
      <button
        class="orange"
        type="button"
      >
        +
      </button>
      <button
        class="span-two"
        type="button"
      >
        0
      </button>
      <button
        type="button"
      >
        .
      </button>
      <button
        class="orange"
        type="button"
      >
        =
      </button>
    </div>
  </div>
</DocumentFragment>
`);
});

test('adds numbers correctly', () => {
  const { getByText, getByTestId } = render(<Calculator />);
  const display = getByTestId('current-operand');

  fireEvent.click(getByText('1'));
  fireEvent.click(getByText('+'));
  fireEvent.click(getByText('2'));
  fireEvent.click(getByText('='));

  expect(display.textContent).toBe('3');
});

test('performs division correctly', () => {
  const { getByText, getByTestId } = render(<Calculator />);
  const display = getByTestId('current-operand');

  fireEvent.click(getByText('8'));
  fireEvent.click(getByText('รท'));
  fireEvent.click(getByText('2'));
  fireEvent.click(getByText('='));

  expect(display.textContent).toBe('4');
});
