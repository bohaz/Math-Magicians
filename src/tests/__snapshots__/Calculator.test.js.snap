// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Calculator correctly 1`] = `
<DocumentFragment>
  <div
    class="calculator-container"
  >
    <h2
      class="clalculator-title"
    >
      Let's do some math!
    </h2>
    <div
      class="calculator-grid"
    >
      <div
        class="output"
      >
        <div
          class="current-operand"
        >
          0
        </div>
      </div>
      <button
        type="button"
      >
        AC
      </button>
      <button
        type="button"
      >
        +/-
      </button>
      <button
        type="button"
      >
        %
      </button>
      <button
        class="orange"
        type="button"
      >
        รท
      </button>
      <button
        type="button"
      >
        7
      </button>
      <button
        type="button"
      >
        8
      </button>
      <button
        type="button"
      >
        9
      </button>
      <button
        class="orange"
        type="button"
      >
        x
      </button>
      <button
        type="button"
      >
        4
      </button>
      <button
        type="button"
      >
        5
      </button>
      <button
        type="button"
      >
        6
      </button>
      <button
        class="orange"
        type="button"
      >
        -
      </button>
      <button
        type="button"
      >
        1
      </button>
      <button
        type="button"
      >
        2
      </button>
      <button
        type="button"
      >
        3
      </button>
      <button
        class="orange"
        type="button"
      >
        +
      </button>
      <button
        class="span-two"
        type="button"
      >
        0
      </button>
      <button
        type="button"
      >
        .
      </button>
      <button
        class="orange"
        type="button"
      >
        =
      </button>
    </div>
  </div>
</DocumentFragment>
`;
